components/modals/CreateCampaignStages/CampaignStageThree.tsx

```diff
@@ -103,20 +156,87 @@ export const CampaignStepThree = (
             />
 
             <Text>Purpose of the Stage</Text>
-            <TextInput
-              style={styles.input3}
-              value={stage.purpose}
-              onChangeText={(text) => handleStageChange(index, "purpose", text)}
-            />
+            <View
+              style={{
+                flexDirection: "row",
+                alignItems: "center",
+                position: "relative",
+              }}
+            >
+              <TextInput
+                style={styles.input3}
+                value={stage.purpose}
+                onChangeText={(text) =>
+                  handleStageChange(index, "purpose", text)
+                }
+              />
+              <TouchableOpacity
+                onPress={() =>
+                  Platform.OS === "web"
+                    ? openModal(
+                        "Edit Stage Purpose",
+                        "Enter Stage Purpose",
+                        stage.purpose,
+                        (value) => handleStageChange(index, "purpose", value)
+                      )
+                    : router.push({
+                        pathname: "/(main)/(screens)/(campaigns)/textbox-page",
+                        params: {
+                          title: "Edit Stage Purpose " + index,
+                          value: stage.purpose,
+                          path: "/campaigns/create",
+                        },
+                      })
+                }
+                style={styles.editIcon}
+              >
+                <Ionicons name="pencil" size={24} color="black" />
+              </TouchableOpacity>
+            </View>
 
             <Text>Example Conversations</Text>
-            <TextInput
-              style={styles.input3}
-              value={stage.exampleConversations}
-              onChangeText={(text) =>
-                handleStageChange(index, "exampleConversations", text)
-              }
-            />
+            <View
+              style={{
+                flexDirection: "row",
+                alignItems: "center",
+                position: "relative",
+              }}
+            >
+              <TextInput
+                style={styles.input3}
+                value={stage.exampleConversations}
+                onChangeText={(text) =>
+                  handleStageChange(index, "exampleConversations", text)
+                }
+              />
+              <TouchableOpacity
+                onPress={() =>
+                  Platform.OS === "web"
```

Comment:
I saw this 5-6 times, may be we can refactor this and make a reusable utility or something?